cli/src/semgrep/app/scans.py

```diff
@@ -290,6 +331,16 @@ def report_findings(
             f"{state.env.semgrep_url}/api/agent/scans/{self.scan_id}/findings_and_ignores",
             json=findings_and_ignores,
         )
+        # TODO: delete this once all on-prem app instances are gone
+        if response.status_code == 404:
```

Comment:
Can we also add a condition here that state.env.semgrep_url is not https://semgrep.dev so we only fall back to this on VPC